subprojects/composite-builds/src/integTest/groovy/org/gradle/integtests/composite/CompositeBuildConfigurationAttributesResolveIntegrationTest.groovy

```diff
@@ -94,10 +94,12 @@ class CompositeBuildConfigurationAttributesResolveIntegrationTest extends Abstra
                 bar.attributes { attribute(buildType, 'release'); attribute(flavor, 'free') }
             }
             task fooJar(type: Jar) {
-               baseName = 'c-foo'
+                baseName = 'c-foo'
+                destinationDir = projectDir
```

Comment:
Couldn't this be in a `setup()` block? Same for most of the other tests. Also, why not apply the 'base' plugin instead?

Reply:
I generally tried to apply the `base` plugin where possible. However, there were lots of tests that defined conflicting configurations etc. so I eventually changed my strategy to using `destinationDir`. Moreover, that makes it more explicit. Someone looking at the test later might not know why the `base` plugin was applied. As for the duplication, there certainly is potential for refactoring in some of these test classes. I'll send a follow-up PR.